File: EventBa.Services/Database/Notification.cs
using System;
using System.Collections.Generic;
using EventBa.Models.Enums;

namespace EventBa.Services.Database;

public partial class Notification
{
    public Guid Id { get; set; }

    public Guid? UserId { get; set; }

    public Guid? EventId { get; set; }

    public bool IsSystemNotification { get; set; }

    public string Title { get; set; } = null!;

    public string Content { get; set; } = null!;

    public bool IsImportant { get; set; }

    public NotificationStatus Status { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual Event? Event { get; set; }

    public virtual User? User { get; set; }
}

File: UI/eventba_mobile/lib/models/notification/notification.dart
import 'package:json_annotation/json_annotation.dart';

import '../enums/notification_status.dart';

part 'notification.g.dart';

@JsonSerializable()
class Notification {
  final String id;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String? eventId;
  final bool isSystemNotification;
  final String title;
  final String content;
  final bool isImportant;
  final NotificationStatus status;

  @JsonKey(ignore: true)
  bool get isRead => status == NotificationStatus.Read;

  Notification({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    this.eventId,
    required this.isSystemNotification,
    required this.title,
    required this.content,
    required this.isImportant,
    required this.status,
  });

  factory Notification.fromJson(Map<String, dynamic> json) =>
      _$NotificationFromJson(json);

  Map<String, dynamic> toJson() => _$NotificationToJson(this);
}
